datasource/ethereum/src/block_stream.rs

```diff
@@ -510,44 +577,64 @@ where
-                        // processed.
-                        subgraph_ptr = (&descendant_block).into();
-
-                        Ok(descendant_block)
-                    })) as Box<Stream<Item = _, Error = _> + Send>,
+                    Box::new(descendant_blocks
+                             .and_then(move |descendant_block| {
+                                 // First, check if there are blocks between subgraph_ptr and
+                                 // descendant_block.
+                                 let descendant_parent_ptr = EthereumBlockPointer::to_parent(&descendant_block.ethereum_block);
+                                 if subgraph_ptr != descendant_parent_ptr {
+                                     // descendant_block is not a direct child.
+                                     // Therefore, there are blocks that are irrelevant to this subgraph
+                                     // that we can skip.
+                                     debug!(
+                                         ctx.logger,
+                                         "Skipping {} block(s) with no relevant triggers...",
+                                         descendant_parent_ptr.number - subgraph_ptr.number
+                                     );
+
+                                     // Update subgraph_ptr in store to skip the irrelevant blocks.
+                                     ctx.subgraph_store.set_block_ptr_with_no_changes(
+                                         ctx.subgraph_id.clone(),
+                                         subgraph_ptr,
+                                         descendant_parent_ptr,
+                                     )?;
+                                 }
+
+                                 // Update our copy of the subgraph ptr to reflect the
+                                 // value it will have after descendant_block is
+                                 // processed.
+                                 subgraph_ptr = (&descendant_block.ethereum_block).into();
+
+                                 Ok(descendant_block)
+                             })
+                             .and_then(move |descendant_block| {
+                                 let triggers = match parse_triggers(
+                                     log_filter_opt.clone(),
+                                     call_filter_opt.clone(),
+                                     block_filter_opt.clone(),
+                                     &descendant_block,
+                                 ) {
+                                     Ok(triggers) => triggers,
+                                     Err(err) => return future::err(err)
+                                 };
+                                 future::ok(EthereumBlockWithTriggers {
+                                     ethereum_block: descendant_block.ethereum_block,
+                                     triggers: triggers,
```

Comment:
This can just be `triggers` instead of `triggers: triggers`.

I'd also merge the two expressions here into one:

```rust
future::result(
    parse_triggers(
        log_filter_opt.clone(),
        ...
    ).map(move |triggers| EthereumBlockWithTriggers {
        ethereum_block: descendant_block.ethereum_block,
        triggers,
    })
)
```

(I haven't tested this but I think it should work.)

Reply:
It worked